app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -54,7 +54,32 @@ def internal_networks
       network_client.networks.all.select { |net| !net.router_external }
     end
 
+    def image_size(image_id)
+      client.get_image_details(image_id).body['image']['minDisk']
+    end
+
+    def boot_from_volume(args = {})
+      vm_name = args[:name]
+      args[:size_gb] = image_size(args[:image_ref]) if args[:size_gb].blank?
+      boot_vol = volume_client.volumes.create( { :display_name => "volume-#{vm_name}", :volumeType => "Volume", :size => args[:size_gb], :imageRef => args[:image_ref] } )
+      @boot_vol_id = boot_vol.id.tr('"', '')
+      boot_vol.wait_for { status == 'available'  }
+      args[:block_device_mapping] = [ {
+        :api_ver => "v2",
```

Comment:
@eLobato @jslatten I think the argument should be block_device_mapping_v2 and not block_device_mapping (and it works for me fine on Juno, if I change it to block_device_mapping_v2).